### ISSUE TITLE
PLEXTOR - CD-R PX-W2410A does not support D8

### ISSUE THREAD

**Issue by smesgr9000** [superg/redumper#178]
I shot myself a CD-R PX-W2410A but the device doesn't seem to support **D8**.
> drive: PLEXTOR - CD-R PX-W2410A (revision level: 1.04, vendor specific: 04/25/02 19:00)

It either generates alot of
> SCSI error (SC: CHECK CONDITION, SK: ILLEGAL REQUEST, ASC: INVALID FIELD IN CDB)

with C2 or without C2
> SCSI error (SC: CHECK CONDITION, SK: ILLEGAL REQUEST, ASC: ILLEGAL MODE FOR THIS TRACK)

See attachments

it works fine with **BE**
> drive configuration: GENERIC (read offset: +98, C2 shift: 294, pre-gap start: -75, read method: BE, sector order: DATA_C2_SUB)

[drive_test.log](https://github.com/user-attachments/files/16447156/drive_test.log)
[drive_test1.log](https://github.com/user-attachments/files/16447170/drive_test1.log)

**Comment by superg** [comment]
Huh, this is an interesting information but from your log I see that leadin was read and it's read with D8.
Can you try to exclude C2 like this:
`redumper --verbose --overwrite --drive-sector-order=DATA_SUB --image-name=drive_test`
and see if it works?

**Comment by smesgr9000** [comment]
Isn't it what drive_test1.log is doing?

**Comment by smesgr9000** [comment]
sorry took me some time.

I have tested this drive now:
Linux -  LogiLink AU0028A USB-IDE-Adapter -> same issue
Windows 7 - LogiLink AU0028A USB-IDE-Adapter -> same issue
Linux - offbrand chinese JMB363 PCIe-IDE-Adapter -> same issue

**Comment by Deterous** [comment]
> I have tested this drive now: Linux - LogiLink AU0028A USB-IDE-Adapter -> same issue Windows 7 - LogiLink AU0028A USB-IDE-Adapter -> same issue Linux - offbrand chinese JMB363 PCIe-IDE-Adapter -> same issue

It may be an issue with your USB adaptor, which has caused issues with getting Plextors to work properly in the past.

**Comment by smesgr9000** [comment]
the JMB363 is no USB adapter It is as stated an PCIe card

**Comment by smesgr9000** [comment]
i tried to run the software on my old Sockel 7 system. But could get redumper build for on Debian i386. cmake fails at the linker stage with out of memory (even through the build system has 16GB RAM and the linker never uses more than 1.5GB)

btw the AU0028A uses an IS611 which is as far as I know the gold standard for USB-IDE-Adapter.

**Comment by superg** [comment]
Doesn't precompiled 32bit version work?
I link statically, it should work anywhere.

**Comment by smesgr9000** [comment]
I can't see any Linux precompiled 32bit versions. 
Sadly Windows 98/ME is not supported by the Windows 32bit executable. I will try to install NT or XP on this machine if I have some time. Would be nice to know which Windows is the oldest I give a try. This system is from 1998, XP will already be hard to get going. Thus I tried a Linux distro first. Sadly I have no other IDE based system available at the moment.

**Comment by superg** [comment]
My bad, I misremember things, I thought that I build 32-bit linux binary, but it's only windows 32-bit build that I do.

**Comment by superg** [comment]
The earliest windows you can do is Windows 7 32-bit. Sadly XP will not work.

**Comment by smesgr9000** [comment]
just keep you updated. I have tested the same usb-adapters with an PX-712A now and D8 is working there.

Still try to build a newer system with ide & W7.

**Comment by superg** [comment]
Ultimately I'll acquire that drive here to do some tests and investigate what's going on there. D8 available for lead-in but not available for the main channel is definitely weird, maybe it just requires some special options in SCSI read command.

**Comment by smesgr9000** [comment]
i could borrow my drive, currently it is no use to me

**Comment by superg** [comment]
Yeah thanks for offering that but I don't currently have bandwidth to look into that, still have to complete more important features.
I'll keep that in mind.

**Comment by smesgr9000** [comment]
CD-R PX-W1210A has the same issue. Which drives are actually known to work?

[dump_241228_152956_sg0.log](https://github.com/user-attachments/files/18266488/dump_241228_152956_sg0.log)

**Comment by Deterous** [comment]
http://wiki.redump.org/index.php?title=Optical_Disc_Drive_Compatibility:_CD
W1210A is not on the redump list, only W1210TS is (and it has limitations)

**Comment by Deterous** [comment]
Regarding W2410A and W1210A (and other plextors from similar generation), they will not dump using 0xD8 opcode for pure CD-ROM discs.
These drives support 0xD8 and DATA_SUB_C2 for Audio CDs, and they support 0xD8 and DATA_SUB for mixed mode CDs.

**Comment by superg** [comment]
I've just verified PX-W2410A.
It supports D8 but doesn't support C2 when using D8.
Closing.
